src/cpp/src/tokenizer.cpp

```diff
@@ -117,59 +132,51 @@ class Tokenizer::TokenizerImpl {
 
     TokenizerImpl() = default;
 
-    TokenizerImpl(std::filesystem::path tokenizer_path, const ov::AnyMap& properties)
-        : m_chat_template{chat_template_from_tokenizer_json_if_exists(tokenizer_path)} {
-        ov::Core core;
+    TokenizerImpl(const std::pair<std::shared_ptr<ov::Model>, std::shared_ptr<ov::Model>>& models,  const ov::AnyMap& properties) {
+        auto [ov_tokenizer, ov_detokenizer] = models;
 
-        OPENVINO_ASSERT(tokenizer_path.extension() != ".xml", "'tokenizer_path' parameter should be a path to a dir not a xml file");
-
-        const char* ov_tokenizer_path = getenv(ScopedVar::ENVIRONMENT_VARIABLE_NAME);
-        OPENVINO_ASSERT(ov_tokenizer_path, "openvino_tokenizers path is not set");
-        core.add_extension(ov_tokenizer_path);
-
-        read_config(tokenizer_path);
-        read_special_tokens_map(tokenizer_path);
-
-        // Try to read tokenizer_config if some token ids or token str are not defined.
-        read_tokenizer_config_if_necessary(tokenizer_path);
-
-        auto device = "CPU"; // currently openvino_tokenizer supports only CPU
-        auto ov_tokenizer = core.read_model(tokenizer_path / "openvino_tokenizer.xml");
         m_older_than_24_5 = ov_tokenizer->get_rt_info().count("openvino_tokenizers_version") != 1;
-        
-        ov::pass::Manager manager;
-        manager.register_pass<MakeCombineSegmentsSatateful>();
-        manager.run_passes(ov_tokenizer);
-        
-        m_tokenizer = core.compile_model(ov_tokenizer, device, properties);
-        if (std::filesystem::exists(tokenizer_path / "openvino_detokenizer.xml")) {
-            m_detokenizer = core.compile_model(tokenizer_path / "openvino_detokenizer.xml", device, properties);
+        auto core = get_core_singleton();
+        std::string device = "CPU"; // only CPU is supported for now
+        if (ov_tokenizer) {
```

Comment:
you need to add asserts to `encode` that tokenizer is available (currently, it's done only for detokenizer)

Reply:
done